From 3826f62c0ad3950780afd1f1bb66d13444bd4b6b Mon Sep 17 00:00:00 2001
From: Max Horn <max@quendi.de>
Date: Thu, 5 Oct 2023 23:35:06 +0200
Subject: [PATCH] Fix compiler warning on macOS with GCC

With GCC 13 on macOS I was getting this warning:

    warning: implicit declaration of function '_NSGetExecutablePath'
---
 src/main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/main.c b/src/main.c
index 50f49952f..d07bb2558 100644
--- a/src/main.c
+++ b/src/main.c
@@ -24,6 +24,12 @@
 #include <string.h>
 #include <unistd.h>
 
+#if defined(__APPLE__) && defined(__MACH__)
+// Workaround: TRUE / FALSE are also defined by the macOS Mach-O headers
+#define ENUM_DYLD_BOOL
+#include <mach-o/dyld.h>
+#endif
+
 extern int realmain(int argc, char * argv[]);
 
 /****************************************************************************
-- 
2.42.0

